ledger-api/grpc-definitions/com/daml/ledger/api/v1/ledger_identity_service.proto

```diff
@@ -9,8 +9,9 @@ option java_outer_classname = "LedgerIdentityServiceOuterClass";
 option java_package = "com.daml.ledger.api.v1";
 option csharp_namespace = "Com.Daml.Ledger.Api.V1";
 
+// DEPRECATED: This service is now deprecated and ledger identity string is optional for all Ledger API requests.
+//
```

Comment:
@pbatko-da : I'd suggest to also add `option deprecated = true` annotations to the service and its RPC; and `[deprecated = true]` annotations to its messages as outlined here: https://stackoverflow.com/questions/43275832/how-to-mark-rpc-as-deprecated

This should transport the deprecation info via compiler annotations on generated code, e.g., Java's `@Deprecated` annotations. 

AFAIK, this is the first time we are deprecating a service and service methods. Have a look for what breaks.

Reply:
(fyi  @stefanobaghino-da)

Added deprecated options in proto file and `@nowarn` annotations to selectively suppress compiler warnings.

FYI This is the result of adding the deprecation options in `com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc` class: annotations like `@scala.deprecated(message = "Marked as deprecated in proto file", since = "")` have appeared:

```
package com.daml.ledger.api.v1.ledger_identity_service
@scala.deprecated(message = "Marked as deprecated in proto file", since = "")
object LedgerIdentityServiceGrpc extends scala.AnyRef {
  @scala.deprecated(message = "Marked as deprecated in proto file", since = "")
  val METHOD_GET_LEDGER_IDENTITY : io.grpc.MethodDescriptor[com.daml.ledger.api.v1.ledger_identity_service.GetLedgerIdentityRequest, com.daml.ledger.api.v1.ledger_identity_service.GetLedgerIdentityResponse] = { /* compiled code */ }
  val SERVICE : io.grpc.ServiceDescriptor = { /* compiled code */ }
  trait LedgerIdentityService extends scala.AnyRef with scalapb.grpc.AbstractService {
    override def serviceCompanion : com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc.LedgerIdentityService = { /* compiled code */ }
    @scala.deprecated(message = "Marked as deprecated in proto file", since = "")
    def getLedgerIdentity(request : com.daml.ledger.api.v1.ledger_identity_service.GetLedgerIdentityRequest) : scala.concurrent.Future[com.daml.ledger.api.v1.ledger_identity_service.GetLedgerIdentityResponse]
  }
  object LedgerIdentityService extends scalapb.grpc.ServiceCompanion[com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc.LedgerIdentityService] {
    implicit def serviceCompanion : scalapb.grpc.ServiceCompanion[com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc.LedgerIdentityService] = { /* compiled code */ }
    def javaDescriptor : com.google.protobuf.Descriptors.ServiceDescriptor = { /* compiled code */ }
    def scalaDescriptor : scalapb.descriptors.ServiceDescriptor = { /* compiled code */ }
    def bindService(serviceImpl : com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc.LedgerIdentityService, executionContext : scala.concurrent.ExecutionContext) : io.grpc.ServerServiceDefinition = { /* compiled code */ }
  }
  trait LedgerIdentityServiceBlockingClient extends scala.AnyRef {
    def serviceCompanion : com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc.LedgerIdentityService = { /* compiled code */ }
    @scala.deprecated(message = "Marked as deprecated in proto file", since = "")
    def getLedgerIdentity(request : com.daml.ledger.api.v1.ledger_identity_service.GetLedgerIdentityRequest) : com.daml.ledger.api.v1.ledger_identity_service.GetLedgerIdentityResponse
  }
  class LedgerIdentityServiceBlockingStub(channel : io.grpc.Channel, options : io.grpc.CallOptions = { /* compiled code */ }) extends io.grpc.stub.AbstractStub[com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc.LedgerIdentityServiceBlockingStub] with com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc.LedgerIdentityServiceBlockingClient {
    @scala.deprecated(message = "Marked as deprecated in proto file", since = "")
    override def getLedgerIdentity(request : com.daml.ledger.api.v1.ledger_identity_service.GetLedgerIdentityRequest) : com.daml.ledger.api.v1.ledger_identity_service.GetLedgerIdentityResponse = { /* compiled code */ }
    override def build(channel : io.grpc.Channel, options : io.grpc.CallOptions) : com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc.LedgerIdentityServiceBlockingStub = { /* compiled code */ }
  }
  class LedgerIdentityServiceStub(channel : io.grpc.Channel, options : io.grpc.CallOptions = { /* compiled code */ }) extends io.grpc.stub.AbstractStub[com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc.LedgerIdentityServiceStub] with com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc.LedgerIdentityService {
    @scala.deprecated(message = "Marked as deprecated in proto file", since = "")
    override def getLedgerIdentity(request : com.daml.ledger.api.v1.ledger_identity_service.GetLedgerIdentityRequest) : scala.concurrent.Future[com.daml.ledger.api.v1.ledger_identity_service.GetLedgerIdentityResponse] = { /* compiled code */ }
    override def build(channel : io.grpc.Channel, options : io.grpc.CallOptions) : com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc.LedgerIdentityServiceStub = { /* compiled code */ }
  }
  def bindService(serviceImpl : com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc.LedgerIdentityService, executionContext : scala.concurrent.ExecutionContext) : io.grpc.ServerServiceDefinition = { /* compiled code */ }
  def blockingStub(channel : io.grpc.Channel) : com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc.LedgerIdentityServiceBlockingStub = { /* compiled code */ }
  def stub(channel : io.grpc.Channel) : com.daml.ledger.api.v1.ledger_identity_service.LedgerIdentityServiceGrpc.LedgerIdentityServiceStub = { /* compiled code */ }
  def javaDescriptor : com.google.protobuf.Descriptors.ServiceDescriptor = { /* compiled code */ }
}
```

Reply:
Thank you @pbatko-da . What would it take to fill the `since = ""` with `since = 2.0`?

Reply:
Thanks. As mentioned in a comment, please be as specific as possible:
- if feasible, annotate with `@nowarn` at the expression level
- in any case, for deprecation warnings, don't use `msg=deprecated` but `cat=deprecation&origin=<regex-matching-the-deprecation-origin>`

Here is an example of a rather broad exclusion:

https://github.com/digital-asset/daml/blob/0f4e952911811d083b08e32ec2d70db7ab581afa/language-support/java/bindings/src/test/scala/com/daml/ledger/javaapi/data/SubmitCommandsRequestSpec.scala#L16-L17

And here one of a more specific one, which is preferable:

https://github.com/digital-asset/daml/blob/0f4e952911811d083b08e32ec2d70db7ab581afa/language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/DamlLedgerClientTest.scala#L45-L50

Reply:
@meiersi-da 
> What would it take to fill the `since = ""` with `since = 2.0`?

I didn't find any mention about `since` attribute in the web or in the docs.
(Proto docs on `deprecated` option: https://developers.google.com/protocol-buffers/docs/overview?hl=en#:~:text=deprecated%20(field%20option)%3A)

@stefanobaghino-da 
> As mentioned in a comment, please be as specific as possible:

Ok, thanks for the examples! Changing

Reply:
OK. That would probably require a deeper dive into our proto -> scala code generatoin pipeline. Let's skip it for now.